src/image.c

```diff
@@ -31,27 +31,45 @@ typedef enum {
 #undef _
 } twin_image_format_t;
 
-/* FIXME: Check the header of the given images to determine the supported image
- * formats instead of parsing the filename without checking its content.
- */
+
 static twin_image_format_t image_type_from_name(const char *path)
 {
     twin_image_format_t type = IMAGE_TYPE_unknown;
-    const char *extname = strrchr(path, '.');
-    if (!extname)
+    FILE *file = fopen(path, "rb");
+
+    if (!file) {
+        perror("Failed to open %s");
         return IMAGE_TYPE_unknown;
+    }
+
+    uint8_t header[8];
+    size_t bytes_read = fread(header, 1, sizeof(header), file);
+    fclose(file);
+
+    if (bytes_read < 8) {
+        return IMAGE_TYPE_unknown;
+    }
+
 #if LOADER_HAS(PNG)
-    else if (!strcasecmp(extname, ".png")) {
+#if __BYTE_ORDER == __BIG_ENDIAN
+    uint8_t png_header[8] = {0x89, 0x50, 0x4E, 0x47, 0x0D, 0x0A, 0x1A, 0x0A};
```

Comment:
I think `static const` should apply to 'jpg_header' as well ?
`static const uint8_t jpg_header[3] = {0xFF, 0xD8, 0xFF};`

Reply:
Yes. You should move the declarations prior to the if-else statements.

Reply:
For the sake of naming scheme, use `header_png` rather than `png_header`.

Reply:
Like the below
![image](https://github.com/user-attachments/assets/c605ff48-a54e-4fea-83dc-a78b8c8fe900)

![image](https://github.com/user-attachments/assets/b882bd35-38d8-4574-93c9-2e6029647fdb)

Reply:
Never put screenshots which contains plain text only.

Reply:
```
/* Header file format of supported image */
#if __BYTE_ORDER == __BIG_ENDIAN
static const uint8_t header_png[8] = {0x89, 0x50, 0x4E, 0x47,
                                      0x0D, 0x0A, 0x1A, 0x0A};
#else
static const uint8_t header_png[8] = {0x0A, 0x1A, 0x0A, 0x0D,
                                      0x47, 0x4E, 0x50, 0x89};
#endif

static const uint8_t header_jpg[3] = {0xFF, 0xD8, 0xFF};
```


```
#if LOADER_HAS(PNG)
    else if (!memcmp(header, header_png, sizeof(header_png))) {
        type = IMAGE_TYPE_png;
    }
#endif

#if LOADER_HAS(JPEG)
    else if (!memcmp(header, header_jpg, sizeof(header_jpg))) {
        type = IMAGE_TYPE_jpeg;
    }
#endif

    /* otherwise, unsupported format */
    return type;
}
```
After modification like this.